nextcloud_integration/models.py

```diff
@@ -51,43 +51,53 @@ def __str__(self):
         return f"Folder: {self.display_name}"
 
 
-class NCFile(models.Model):
+class SquireNextCloudFile(models.Model):
     """ Represents a file on the nextcloud storage """
-    display_name = models.CharField(help_text="Name displayed in Squire", max_length=32)
+    display_name = models.CharField(help_text="Name displayed in Squire", max_length=64)
     description = models.CharField(max_length=256, blank=True)
     file_name = models.CharField(help_text="The file name", max_length=64, blank=True, default=None)
-    folder = models.ForeignKey(NCFolder, on_delete=models.CASCADE, related_name="files")
+    folder = models.ForeignKey(SquireNextCloudFolder, on_delete=models.CASCADE, related_name="files")
     slug = models.SlugField(blank=True)
     file: NextCloudFolder = None  # Used to translate Nextcloud folder contents
 
     is_missing = models.BooleanField(default=False) # Whether the file is non-existant on nextcloud
+    CONNECTION_NEXTCLOUD_SYNC = "NcS"
+    CONNECTION_SQUIRE_UPLOAD = "SqU"
     connection = models.CharField(max_length=3, choices=[
-        ("NcS", "Synched through file on Nextcloud"),
-        ("SqU", "Uploaded through Squire")
+        (CONNECTION_NEXTCLOUD_SYNC, "Synched through file on Nextcloud"),
+        (CONNECTION_SQUIRE_UPLOAD, "Uploaded through Squire")
     ]) # Defines how the connection occured
 
     class Meta:
-        # Set the default permissions. Each item has a couple of addiotional default permissions
+        # Set the default permissions. Each item has a couple of additional default permissions
         default_permissions = ('add', 'change', 'delete', 'view',
-                               'synch',)
+                               'sync',)
 
     def __init__(self, *args, **kwargs):
-        super(NCFile, self).__init__(*args, **kwargs)
+        super(SquireNextCloudFile, self).__init__(*args, **kwargs)
         if self.id:
             self.file = NextCloudFile(path=self.path)
 
     def save(self, **kwargs):
         if self.file and (self.file_name == "" or self.file_name is None):
             self.file_name = self.file.name
 
-        if self.slug is None or self.slug == "":
-            self.slug = slugify(self.file_name.replace('.', '_'))
+        # Cleaning is not guaranteed when saving
+        self._set_slug()
 
-        return super(NCFile, self).save(**kwargs)
+        return super(SquireNextCloudFile, self).save(**kwargs)
 
     def clean(self):
+        self._set_slug()
         if self.file is None and (self.file_name is None or self.file_name == ""):
-            raise ValidationError("Either file or filename should be defined")
+            raise ValidationError("Either file or filename should be defined", code="no_file_name_linked")
+
+        if SquireNextCloudFile.objects.filter(folder=self.folder, slug=self.slug).exclude(id=self.id).exists():
```

Comment:
Fine as is, but this is essentially a [unique_together](https://docs.djangoproject.com/en/3.2/ref/models/options/#unique-together) on the slug and folder